catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/MetacardTypeRegister.java

```diff
@@ -34,19 +34,19 @@
 
     private Map<String, ServiceRegistration> metacardTypeServiceRegistrations = new HashMap<>();
 
-    private List<SaxEventHandlerFactory> saxEventHandlerFactories;
-
-    private List<String> saxEventHandlerConfiguration;
+    private List<SaxEventHandlerFactory> saxEventHandlerFactories = new ArrayList<>();
 
     private DynamicMetacardType metacardType;
 
     private String id = "DEFAULT_ID";
 
     public void bind(SaxEventHandlerFactory saxEventHandlerFactory) {
```

Comment:
You'll now need to synchronize your reads/writes to the `saxEventHandlerFactories` list (and calls to `setMetacardType`. Simplest to just sync the `bind`, `unbind`, and `setMetacardType` methods. You could use a lock instead but that's probably overkill.